### ISSUE TITLE
Push notifications don't work

### ISSUE THREAD

**Issue by carbonin** [ManageIQ/manageiq-pods#517]
In the ops UI push notifications don't work and I'm seeing an error in the browser console related to websockets (I assume that's related).

![image](https://user-images.githubusercontent.com/12697904/83069803-7fbe1d80-a038-11ea-84fc-c838f1f4b2db.png)

**Comment by carbonin** [comment]
So first pass on this is that the service name changed from `websocket` to `remote-console` so [this config](https://github.com/ManageIQ/manageiq-pods/blob/3284ac0e16237009ef389b5261f7b309e3f53809/manageiq-operator/pkg/helpers/miq-components/httpd_conf.go#L37-L38) is clearly never going to work.

In master we have this in the httpd logs:
```
[Thu Jun 04 18:25:45.161400 2020] [proxy:error] [pid 227:tid 140261111179008] [client 10.128.0.1:47610] AH00898: DNS lookup failure for: websocket returned by /ws/notifications
10.128.0.1 - - [04/Jun/2020:18:25:45 +0000] "GET /ws/notifications HTTP/1.1" 502 412
10.128.0.1 - - [04/Jun/2020:18:25:45 +0000] "GET /ws/notifications HTTP/1.1" 502 412
```

Which checks out with the services not matching.

**Comment by carbonin** [comment]
In an effort to make changes similar to https://github.com/ManageIQ/manageiq-appliance/pull/219 I made the following change:

```patch
diff --git a/manageiq-operator/pkg/helpers/miq-components/httpd_conf.go b/manageiq-operator/pkg/helpers/miq-components/httpd_conf.go
index e9e41f6..f62348b 100644
--- a/manageiq-operator/pkg/helpers/miq-components/httpd_conf.go
+++ b/manageiq-operator/pkg/helpers/miq-components/httpd_conf.go
@@ -31,13 +31,8 @@ Options SymLinksIfOwnerMatch
 
   ProxyPreserveHost on
 
-  RewriteCond %{REQUEST_URI}     ^/ws        [NC]
-  RewriteCond %{HTTP:UPGRADE}    ^websocket$ [NC]
-  RewriteCond %{HTTP:CONNECTION} ^Upgrade$   [NC]
-  RewriteRule .* ws://websocket:3000%{REQUEST_URI}  [P,QSA,L]
-  ProxyPassReverse /ws ws://websocket:3000/ws
-
   RewriteCond %{REQUEST_URI} !^/api
+  RewriteCond %{REQUEST_URI} !^/ws
 
   # For httpd, some ErrorDocuments must by served by the httpd pod
   RewriteCond %{REQUEST_URI} !^/proxy_pages
@@ -54,6 +49,9 @@ Options SymLinksIfOwnerMatch
   ProxyPass /api http://web-service:3000/api
   ProxyPassReverse /api http://web-service:3000/api
 
+  ProxyPass /ws/notifications ws://ui:3000/ws/notifications
+  ProxyPassReverse /ws/notifications ws://ui:3000/ws/notifications
+
   # Ensures httpd stdout/stderr are seen by 'docker logs'.
   ErrorLog  "| /usr/bin/tee /proc/1/fd/2 /var/log/httpd/error_log"
   CustomLog "| /usr/bin/tee /proc/1/fd/1 /var/log/httpd/access_log" common

```

This resulted in 404 rather than 502, which I guess is an improvement?
Here's the console screenshot:
![image](https://user-images.githubusercontent.com/12697904/83798190-e31c0100-a671-11ea-9964-9f7c676c59c1.png)

And here's the logs from the UI worker:
```
Started GET "/ws/notifications" for 127.0.0.1 at 2020-06-04 18:42:53 +0000
Started GET "/ws/notifications/"[non-WebSocket] for 10.10.119.45 at 2020-06-04 18:42:53 +0000"
Failed to upgrade to WebSocket (REQUEST_METHOD: GET, HTTP_CONNECTION: close, HTTP_UPGRADE: )
Finished "/ws/notifications/"[non-WebSocket] for 10.10.119.45 at 2020-06-04 18:42:53 +0000
```

**Comment by carbonin** [comment]
At this point I'm a bit out of my depth. I'm not sure what websockets need to "upgrade" correctly.

@jrafanie @skateman can either of you lend a hand here?

**Comment by carbonin** [comment]
Also added this to the UI worker's httpd config, still the same result:
```patch
diff --git a/images/manageiq-ui-worker/container-assets/manageiq-http.conf b/images/manageiq-ui-worker/container-assets/manageiq-http.conf
index 1232310..31e6f4e 100644
--- a/images/manageiq-ui-worker/container-assets/manageiq-http.conf
+++ b/images/manageiq-ui-worker/container-assets/manageiq-http.conf
@@ -10,6 +10,9 @@ Listen 3000
 <VirtualHost *:3000>
   DocumentRoot /var/www/miq/vmdb/public
 
+  ProxyPass /ws/notifications ws://localhost:3001/ws/notifications
+  ProxyPassReverse /ws/notifications ws://localhost:3001/ws/notifications
+
   RewriteRule ^/ui/service(?!/(assets|images|img|styles|js|fonts|vendor|gettext)) /ui/service/index.html [L]
   RewriteCond %{REQUEST_URI} !^/proxy_pages
   RewriteCond %{DOCUMENT_ROOT}/%{REQUEST_FILENAME} !-f
```

**Comment by skateman** [comment]
The `remote-console` service is not related to the notifications, it runs and should run as a separate worker. The notifications are part of the rails server, so the two things aren't really related to each other.

Is there a way I can play around with this easily?

**Comment by skateman** [comment]
I tried to deploy this on my computer, but the scheduler can't even get to starting the DB :(

**Comment by carbonin** [comment]
Yeah, don't worry about it I think I have it fixed in #535.

#536 might be more difficult though, not sure if it will need more than just adding some redirect/proxy in the httpd conf.
